src/preferences.html

```diff
@@ -0,0 +1,219 @@
+
+    if (notificationMinutes.value === '' || notificationMinutes.value === '0') {
+      notificationMinutes.value = '00';
+    }
+
+    githubUsername.addEventListener('keypress', () => {
+      clearTimeout(typingTimer);
+      if(isInputEmpty(githubUsername)) {
+        return;
+      }
+      githubUsername.parentElement.classList.add('is-loading');
+      typingTimer = setTimeout(() => {
+        ipcRenderer.send('setUsername', githubUsername.value);
+      }, 1000);
+    })
+
+    githubUsername.addEventListener('keydown', () => {
+      clearTimeout(typingTimer);
+    })
+
+    githubSyncInterval.addEventListener('keyup', () => {
+      if (isInputEmpty(githubSyncInterval)) {
+        return;
+      }
+      const syncInterval = parseInt(githubSyncInterval.value, 10);
+      ipcRenderer.send('setSyncInterval', syncInterval);
+    })
+
+    launchAtLoginCheckbox.addEventListener('change', (event) => {
+      ipcRenderer.send('activateLaunchAtLogin', launchAtLoginCheckbox.checked);
+    })
+
+    notificationCheckbox.addEventListener('change', (event) => {
+      if (notificationCheckbox.checked) {
+        notificationHour.disabled = false;
+        notificationMinutes.disabled = false;
+        isInputEmpty(notificationHour);
+        isInputEmpty(notificationMinutes);
+        ipcRenderer.send('activateNotifications', notificationCheckbox.checked);
+      } else {
+        notificationHour.disabled = true;
+        notificationMinutes.disabled = true;
+        notificationHour.classList.remove('is-warning');
+        notificationMinutes.classList.remove('is-warning');
+        ipcRenderer.send('activateNotifications', notificationCheckbox.checked);
+      }
+    })
+
+    notificationHour.addEventListener('keyup', () => {
+      if (isInputEmpty(notificationHour)) {
+        return;
+      }
+      const hours = parseInt(notificationHour.value, 10);
+      if(hours >= 0 && hours <= 23) {
+        ipcRenderer.send('setNotificationTime', {
+          hours: hours,
+          minutes: store.get('notification.minutes')
+        });
+      } else {
+        notificationHour.classList.add('is-warning');
+      }
+    })
+
+    notificationMinutes.addEventListener('keyup', () => {
+      if (isInputEmpty(notificationMinutes)) {
+        return;
+      }
+      const minutes = parseInt(notificationMinutes.value, 10);
+      if (minutes >= 0 && minutes <= 59) {
+        ipcRenderer.send('setNotificationTime', {
+          hours: store.get('notification.hours'),
+          minutes: minutes
+        });
+      } else {
+        notificationMinutes.classList.add('is-warning');
+      }
+    })
+
+    ipcRenderer.on('usernameSet', (event, userExist) => {
+      if (userExist) {
+        console.log('usernameSet : ', store.get('username'));
+        githubUsername.parentElement.classList.remove('is-loading');
+      } else {
+        githubUsername.parentElement.classList.remove('is-loading');
+        githubUsername.classList.add('is-warning');
```

Comment:
I think we could use `is-danger` here since it's an error, I like that we are using `is-warning` when there is no username set though 👍